### PR TITLE
Update with Marco's comments #4

### PR DESCRIPTION
**Pull request by mrdeep1**

### CODE REVIEW

File: draft-ietf-core-new-block.xml [hunk 1]
```diff
@@ -303,8 +303,8 @@
         body).</t>
 
         <t>Q-Block1 Option is useful with the payload-bearing POST, PUT,
-        FETCH, PATCH, and iPATCH requests and their responses (2.01 and
-        2.04).</t>
+        FETCH, PATCH, and iPATCH requests and their responses (2.01, 2.02,
```
Comment (boucadair) [inline draft-ietf-core-new-block.xml:306]: I still don't see how Q-Block1 is useful for 2.02.

File: draft-ietf-core-new-block.xml [hunk 2]
```diff
@@ -446,13 +446,33 @@
...
         <t>2.04 (Changed)<list style="empty">
             <t>This Response Code indicates successful receipt of the entire
             body and the resource was updated. The token used SHOULD be from
             the last received payload. The client should then release all of
             the tokens used for this body.</t>
           </list></t>
 
+        <t>2.05 (Content)<list style="empty">
+            <t>This Response Code indicates successful receipt of the entire
+            FETCH request body (See Section 2 <xref target="RFC8132"></xref>)
```
Comment (boucadair) [inline draft-ietf-core-new-block.xml]: `Section 2 of <xref target="RFC8132">`

File: draft-ietf-core-new-block.xml [hunk 3]
```diff
@@ -1303,6 +1323,86 @@ X<--: Message loss (response)
...
+         |       'Continue' Q-Block2]]
+         +--------->| NON FETCH /path M:0x0d T:0xfd QB2:10/1/1024
+         |<---------+ NON 2.05 M:0x9b T:0xfb O:1335 ET=22 QB2:10/0/1024
+      [[Body has been received]]
+         |   ...    |
+
+]]></artwork>
+          </figure></t>
+
+        <t>If there is Q-Block1 loss, how this is handled is shown in <xref
```
Comment (boucadair) [inline draft-ietf-core-new-block.xml]: s/If there is Q-Block1 loss, how this is handled is shown/If a loss of a block conveying Q-Block1 Option is experienced, then this is handled as shown

File: draft-ietf-core-new-block.xml [hunk 4]
```diff
@@ -1303,6 +1323,86 @@ X<--: Message loss (response)
...
+        target="B3non1"></xref> and <xref target="B3non2"></xref> with FETCH
+        substitued for PUT and 2.04 becomes a 2.05 response.</t> 
+
+        <t>If there is Q-Block2 loss, how this is handled is shown in <xref
```
Comment (boucadair) [inline draft-ietf-core-new-block.xml]: Q-Block2 loss may not parse well. Please reword following the same as suggested for Q-Block1

**Review by boucadair — CHANGES_REQUESTED** [review]

File: draft-ietf-core-new-block.xml (hunk 1 above)
Comment (mrdeep1) [reply draft-ietf-core-new-block.xml:306]: A POST using Q-Block1 can cause a resource to get deleted and this returns a 2.02 code.

Section 5.8.2 rfc7252 states 
````
   If the POST succeeds and results in the target resource being
   deleted, the response SHOULD have a 2.02 (Deleted) Response Code.
   POST is neither safe nor idempotent.
````

File: draft-ietf-core-new-block.txt [hunk 5]
```diff
@@ -1509,11 +1619,76 @@ Internet-Draft      Quick Block-Wise Transfer Options       January 2021
...
+          +--------->| NON FETCH /path M:0x0d T:0xfd QB2:10/1/1024
+          |<---------+ NON 2.05 M:0x9b T:0xfb O:1335 ET=22 QB2:10/0/1024
+       [[Body has been received]]
+          |   ...    |
+
+
+    Figure 12: Example of NON FETCH with Q-Block1 and Q-Block2 Options
+                              (Without Loss)
+
+   If there is Q-Block1 loss, how this is handled is shown in Figure 5
```
Comment (boucadair) [inline draft-ietf-core-new-block.txt]:
```suggestion
   If there is a loss of a message with a Q-Block1 Option, this is handled as shown in Figure 5
```

File: draft-ietf-core-new-block.xml (hunk 2 above)
Comment (mrdeep1) [reply draft-ietf-core-new-block.xml]: Fixed

File: draft-ietf-core-new-block.txt [hunk 6]
```diff
@@ -1509,11 +1619,76 @@ Internet-Draft      Quick Block-Wise Transfer Options       January 2021
...
+   and Figure 6 with FETCH substitued for PUT and 2.04 becomes a 2.05
+   response.
+
+   If there is Q-Block2 loss, how this is handled is shown in Figure 10
```
Comment (boucadair) [inline draft-ietf-core-new-block.txt]:
```suggestion
    If there is a loss of a message with a Q-Block2 Option, this is handled as shown in Figure 10
```

**Review by boucadair — CHANGES_REQUESTED** [review]

File: draft-ietf-core-new-block.xml (hunk 3 above)
Comment (mrdeep1) [reply draft-ietf-core-new-block.xml]: Fixed

File: draft-ietf-core-new-block.xml (hunk 1 above)
Comment (boucadair) [reply draft-ietf-core-new-block.xml:306]: What is probmematic is that the text says:

Q-Block1 Option is **usefu**l with the payload-bearing  ... **and** **their responses**

File: draft-ietf-core-new-block.xml (hunk 4 above)
Comment (mrdeep1) [reply draft-ietf-core-new-block.xml]: Gone with later suggestion

File: draft-ietf-core-new-block.txt (hunk 6 above)
Comment (mrdeep1) [reply draft-ietf-core-new-block.txt]: Updated to this suggestion

File: draft-ietf-core-new-block.txt (hunk 5 above)
Comment (mrdeep1) [reply draft-ietf-core-new-block.txt]: Updated to this suggestion

**Comment by mrdeep1** [comment]
Changes pushed

File: draft-ietf-core-new-block.xml (hunk 1 above)
Comment (mrdeep1) [reply draft-ietf-core-new-block.xml:306]: The response is 2.02 - there does not have to be any data in the response.  The POST body can be large though (but I agree unlikely if the result is the deletion of a resource).  Certainly not used (POST caused delete) in the DOTS case.

I agree that DELETE is more useful than POST to do a deletion of the resource (and not the way I would want to do it, but it is in RFC7252).

Comment (boucadair) [reply draft-ietf-core-new-block.xml:306]: 2.02 has no paylaod, sure. But the text says "Q-Block1 is USEFUL for ... 2.02). I'm not questioning that 2.02 can be returned as per 7252.

Comment (mrdeep1) [reply draft-ietf-core-new-block.xml:306]: OK - POST with Q-Block1 can be used to delete a resource.  Would changing  `is useful` to `can be used` help here?

Comment (boucadair) [reply draft-ietf-core-new-block.xml:306]: Actually, not. 

We don't return any Q-Block1 for 2.01-...-2.05. Saying Q-Block1 is useful for "their responses" would imply that we return an option for those. 

That wording is fine for 7959 as they have:

   CLIENT                                                     SERVER
     |                                                              |
     | CON [MID=1234], POST, /soap, 1:0/1/128 ------>               |
     |                                                              |
     | <------   ACK [MID=1234], 2.31 Continue, 1:0/1/128           |
     |                                                              |
     | CON [MID=1235], POST, /soap, 1:1/1/128 ------>               |
     |                                                              |
     | <------   ACK [MID=1235], 2.31 Continue, 1:1/1/128           |
     |                                                              |
     | CON [MID=1236], POST, /soap, 1:2/0/128, 2:0/0/64 ------>     |
     |                                                              |
     | <------   ACK [MID=1236], 2.04 Changed, 1:2/0/128, 2:0/1/64  |

while we do the following: 

       CoAP        CoAP
       Client      Server
         |          |
         +--------->| CON PUT /path M:0x01 T:0xf0 RT=10 QB1:0/1/1024
         +--------->| CON PUT /path M:0x02 T:0xf1 RT=10 QB1:1/1/1024
         +--------->| CON PUT /path M:0x03 T:0xf2 RT=10 QB1:2/1/1024
         +--------->| CON PUT /path M:0x04 T:0xf3 RT=10 QB1:3/0/1024
         |<---------+ ACK 0.00 M:0x01
         |<---------+ ACK 0.00 M:0x02
         |<---------+ ACK 0.00 M:0x03
         |<---------+ ACK 2.04 M:0x04

Comment (mrdeep1) [reply draft-ietf-core-new-block.xml:306]: OK - understand now.  Will fix accordingly.  Now
````
Q-Block1 Option is useful with the payload-bearing POST, PUT, FETCH, PATCH, and iPATCH requests.
````

**Pull request merged**